Update generator's default value

CaoShuFeng opened the pull request:
Now rbac has upgraded to v1beta1. So "v1beta1" would be a better
default name for users.

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access) 
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`. 
-->
```NONE
```

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/42880)
<!-- Reviewable:end -->

CaoShuFeng committed:
Update generator's default value

Now rbac has upgraded to v1beta1. So "v1beta1" would be a better
default name for users.

liggitt commented on pkg/kubectl/cmd/util/factory_client_access.go:
```diff
@@ -477,8 +477,8 @@ const (
 	SecretForTLSV1GeneratorName            = "secret-for-tls/v1"
 	ConfigMapV1GeneratorName               = "configmap/v1"
-	ClusterRoleBindingV1GeneratorName      = "clusterrolebinding.rbac.authorization.k8s.io/v1alpha1"
-	RoleBindingV1GeneratorName             = "rolebinding.rbac.authorization.k8s.io/v1alpha1"
```
@CaoShuFeng we can't change the name of the generator (that's something users can specify on the CLI and we can't break them). We can add a second generator and change the default, but I'm not sure that's what we want either.

@kubernetes/sig-cli-pr-reviews how do we manage server compatibility crossed with generator versions?

Should we keep a single generator and have make different things based on what the target server supports, or have one generator per API version and select the default generator based on what the target server supports?

kargakis replied:
The second option is what we are doing elsewhere.

liggitt replied:
if that's the path we take, then we need to:
* rename these constants to v1alpha1
* create a second set of constants for v1beta1
* duplicate the generators and tweak them to create v1beta1 objects
* add discovery switching to select the default generator

CaoShuFeng replied:
Thanks for your suggestion. I will keep digging it.
